types/reference.go

```diff
@@ -0,0 +1,36 @@
+/*
+Copyright 2019 The MayaData Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package types
+
+import (
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// Reference contains enough information
+// to let you identify an kubernetes object.
+type Reference struct {
+	// API version of the referent.
```

Comment:
spell check

Reply:
Should we replace referent in the comment with reference

Reply:
:thinking:  Kubernetes is using this for OwnerReference.
Please suggests some alternatives.